sdlib/d/gc/arena.d

```diff
@@ -148,6 +148,43 @@ public:
 		return null;
 	}
 
+	bool resizeLarge(shared(ExtentMap)* emap, Extent* e, size_t size) shared {
+		assert(e !is null, "Extent is null!");
+		assert(e.isLarge(), "Expected a large extent!");
+		assert(e.arenaIndex == index, "Invalid arena index!");
+		assert(isLargeSize(size));
+
+		auto computedPageCount = alignUp(size, PageSize) / PageSize;
+		uint newPages = computedPageCount & uint.max;
+
+		assert(newPages == computedPageCount, "Unexpected page count!");
+
+		uint oldSize = cast(uint) e.size;
+		uint oldPages = oldSize / PageSize;
+
+		// Huge is not supported:
+		if ((e.size > uint.max) || e.isHuge() || (oldPages > PageCount)) {
```

Comment:
Aren't these 3 checks redundant? An extent is going to be huge if it doesn't fit in a huge page - so its size will necesserly be less than 4GB and page count will fit too, no?